### PR TITLE
Product images in front office are now filtered by combination

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Previously, all images are displayed in front office, regardless the pictures choosen for each combination. Now the front office is consistent with the back office |
| Type? | bug fix |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-1023 |
| How to test? | Select images in Product with combinations, then check the front office is consistent with the back office. |

blue_combination

#### Important guidelines
- Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
- Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
- Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

### CODE REVIEW

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -476,12 +476,17 @@ protected function assignAttributesGroups($product_for_template = null)
...
+                        if (!isset($current_cover)) {
+                            $current_cover = array_values($this->context->smarty->tpl_vars['product']->value['images'])[0];
+                        }
 
                         if (is_array($combination_images[$row['id_product_attribute']])) {
                             foreach ($combination_images[$row['id_product_attribute']] as $tmp) {
-                                if ($tmp['id_image'] == $current_cover['id_image']) {
-                                    $combinations[$row['id_product_attribute']]['id_image'] = $id_image = (int) $tmp['id_image'];
-                                    break;
+                                if (isset($current_cover)) {
```
Comment (mickaelandrieu) [inline controllers/front/ProductController.php]: to be removed, dead code

File: src/Adapter/Image/ImageRetriever.php [hunk 2]
```diff
@@ -54,10 +59,12 @@ public function getProductImages(array $product, Language $language)
                 $image['associatedVariants'] = [];
             }
 
-            return $image;
+            if (in_array($productAttributeId, $image['associatedVariants'])) {
```
Comment (mickaelandrieu) [inline src/Adapter/Image/ImageRetriever.php]: the interesting point is here: we shouldnt return all images when we have a product attribute id :)

File: src/Core/Product/AbstractProductPresenter.php [hunk 3]
```diff
@@ -64,8 +64,8 @@ private function fillImages(
         }
 
         if (!isset($presentedProduct['cover'])) {
-            if (isset($presentedProduct['images'][0])) {
-                $presentedProduct['cover'] = $presentedProduct['images'][0];
+            if (count($presentedProduct['images']) > 0) {
+                $presentedProduct['cover'] = array_values($presentedProduct['images'])[0];
```
Comment (mickaelandrieu) [inline src/Core/Product/AbstractProductPresenter.php]: I dont want to rely on an array key, this way we have the first item, regardless the key of table.

**Comment by vincentbz** [comment]
Tested and ok, thanks !
Another PR is needed for the case where the product is offline.

**Comment by mickaelandrieu** [comment]
> Another PR is needed for the case where the product is offline. 

@vincentbz note this is **not at all** related to combinations display :)

File: controllers/front/ProductController.php [hunk 4]
```diff
@@ -476,6 +476,9 @@ protected function assignAttributesGroups($product_for_template = null)
                                 $current_cover = $image;
                             }
                         }
+                        if (!isset($current_cover)) {
+                            $current_cover = array_values($this->context->smarty->tpl_vars['product']->value['images'])[0];
```
Comment (Quetzacoalt91) [inline controllers/front/ProductController.php]: Are you always sure that array has always one value ?

Comment (mickaelandrieu) [inline controllers/front/ProductController.php]: If not, a lot of serious problems will happens (and far before this call, believe me)

**Pull request merged**